### PR TITLE
SIMD Functions for aarch64(ARM64)

### PR DESCRIPTION
**Pull request by weiliw-amz**
*Issue #, if available:*
N/A

*Description of changes:*
Add aarch64 SIMD functions for acceleration.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rofuyu — COMMENTED** [review]
We can consider the following refactor structure:

```
ann/
  hnsw.hpp
  feat_vectors.hpp  (include distances)

  distances.hpp
    do_l2_distance_simd()
    do_dot_product_simd()
    do_dot_product_sparse_simd()
        
  distances_impl/
    default.hpp
      
    x86.hpp (include default.hpp)
      
    aarch64.hpp (include default.hpp)
```

**Comment by weiliw-amz** [comment]
> We can consider the following refactor structure:
> 
> ```
> ann/
>   hnsw.hpp
>   feat_vectors.hpp  (include distances)
> 
>   distances.hpp
>     do_l2_distance_simd()
>     do_dot_product_simd()
>     do_dot_product_sparse_simd()
>         
>   distances_impl/
>     default.hpp
>       
>     x86.hpp (include default.hpp)
>       
>     aarch64.hpp (include default.hpp)
> ```

Modified.

**Comment by weiliw-amz** [comment]
Use baseline default implementation for aarch64 SIMD for now, for simplicity and liability issues.

File: pecos/core/ann/distance.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * and limitations under the License.
+ */
+
+namespace pecos {
+
+namespace ann {
+
+#if defined(__x86_64__) || defined(__amd64__) || defined(_M_IX86) || defined(_M_X64)
+    #include "distance_impl/x86.hpp"
+    #elif defined(__aarch64__)
```
Comment (rofuyu) [inline pecos/core/ann/distance.hpp]: indentation.

File: pecos/core/ann/distance.hpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    #include "distance_impl/aarch64.hpp"
+    #else
```
Comment (rofuyu) [inline pecos/core/ann/distance.hpp]: indentation

File: pecos/core/ann/distance.hpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    inline float do_l2_distance_simd(const float *x, const float *y, size_t len) {
+        return do_l2_distance_simd_default(x, y, len);
+    }
+    inline float do_dot_product_sparse_simd(
+        const size_t s_a, const float * __restrict__ x, const uint32_t * __restrict__ A,
+        const size_t s_b, const float * __restrict__ y, const uint32_t * __restrict__ B) {
+        return do_dot_product_sparse_simd_default(s_a, x, A, s_b, y, B);
+    }
+    #endif
```
Comment (rofuyu) [inline pecos/core/ann/distance.hpp]: indentation

Comment (weiliw-amz) [reply pecos/core/ann/distance.hpp]: resolved

**Review by rofuyu — APPROVED** [review]

**Pull request merged**